File: packages/silo-client-java/src/main/java/in/org/quicko/silo/client/errors/MediaDeleteStalledException.java
package in.org.quicko.silo.client.errors;

/**
 * A {@code 500} where the asset was staged for deletion but the blob store
 * refused to remove its bytes. Not a refusal like {@link MediaInUseException} —
 * it is a storage failure the caller fixes by fixing the blob store, or by
 * running the remedy this exception names.
 */
public class MediaDeleteStalledException extends SiloException {
  private final String remedy;

  public MediaDeleteStalledException(String message, String method, String path, String remedy) {
    super(500, "media_delete_stalled", message, method, path);
    this.remedy = remedy;
  }

  /** What the server says will clear the stall, or {@code ""} when it said nothing. */
  public String remedy() {
    return remedy;
  }
}

File: packages/silo-client-dart/lib/src/errors/media_delete_stalled_exception.dart
import 'silo_exception.dart';

/// A `500`: the asset was staged for deletion and the blob store kept its bytes.
final class MediaDeleteStalledException extends SiloException {
  const MediaDeleteStalledException(String message, String method, String path, this.remedy)
    : super(500, 'media_delete_stalled', message, method, path);

  factory MediaDeleteStalledException.fromWire(String message, String method, String path, Object? details) =>
      MediaDeleteStalledException(message, method, path, switch (details) {
        {'remedy': final String remedy} => remedy,
        _ => '',
      });

  final String remedy;
}
